Logging/LogHandler/FileLogHandler.cs

```diff
@@ -97,6 +158,56 @@ protected override void DisposeSelf()
                 );
 
             m_Writer.WriteLine($"{timestamp}{logLevel}{context}{message}");
+
+            if (m_Rotate && m_Writer.BaseStream.Length > RotateFileSizeLimit)
+            {
+                this.RotateFiles();
+            }
+        }
+
+        private void RotateFiles()
+        {
+            m_Writer.Close();
+            m_Writer = null;
+
+            // Determine how many rotated log files already exist
+            int index = 1;
+            string rotatedFilePath = GetRotatedFilePath(index);
+            while (File.Exists(rotatedFilePath))
```

Comment:
Throughout this class there are a bunch of while loops and they all seem very safe so this is very much a 🚲 comment. 

I'm wondering if we should be explicitly safe and have them all be `for` loops since we know the start and know the upper limit of what could be. Not as clean to look at but guaranteed to never endless loop.

Reply:
That first loop is the problem, since I don't yet know how many rotated log files exist. Even if `RotatedFileCountLimit` is 5, maybe you just lowered it from 10 and there are extra files I need to clean up. At best I could pick an arbitrary upper limit to stop at, but if that limit is reached, apparently there were just that many log files (unless there's a bug in the code I suppose).

The other two could easily be `for` loops though, I'll update those.

Reply:
Now I'm unsure - I think the `for` loops are a bit harder to follow, and I'm not sure they're any less susceptible to infinite-loop-causing bugs. I'll keep whichever version you like more - thoughts?
```cs
// Determine how many rotated log files already exist
int index = 1;
string rotatedFilePath = GetRotatedFilePath(index);
while (File.Exists(rotatedFilePath))
{
    rotatedFilePath = GetRotatedFilePath(++index);
}
int rotatedFileCount = index - 1;

// Delete any files in excess of the file count limit
for (int i = rotatedFileCount; i >= RotateFileCountLimit; i--)
{
    File.Delete(GetRotatedFilePath(i));
    --rotatedFileCount;
}

// Rotate remaining files
for (int i = rotatedFileCount; i >= 0; i--)
{
    File.Move(GetRotatedFilePath(index), GetRotatedFilePath(index + 1));
}
```

Reply:
Yeah in this case, I think the while loops are cleaner and I do think they are unlikely to cause issues. I guess I was more raising the question to the group of whether we should strive for safety in Anvil and where that grey muddy balance is. 

When I see a while loop, an alarm bell goes off but that shouldn't always mean that it can't be a while loop.

Reply:
Rebuilding this with `Directory.GetFiles()` rather than assuming any existing files will be sequential, so no one gets any loops!